### PR TITLE
re-parsing of invalid resources improvements

### PR DESCRIPTION
**Pull request by olensmar**
This PR improves re-parsing of updated resources when they are missing kind/apiVersion

## Changes

-

## Fixes

-

## How to test it

-

## Screenshots

-

## Checklist

- [X] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/redux/services/resource.ts [hunk 1]
```diff
@@ -457,8 +458,17 @@ export function reprocessResources(
         resource.name = `Patch: ${resource.name}`;
       }
 
-      resource.kind = resource.content.kind;
-      resource.version = resource.content.apiVersion;
+      const isKustomziationFile = resource.filePath.endsWith('kustomization.yaml');
```
Comment (devcatalin) [inline src/redux/services/resource.ts:461]: let's have a KUSTOMIZATION_FILE_NAME constant here and use it in the `isKustomizationFile` from `kustomize.ts` as well

File: src/components/molecules/Monaco/Monaco.tsx [hunk 2]
```diff
@@ -191,7 +188,8 @@ const Monaco = (props: {diffSelectedResource: () => void; applySelection: () =>
     if (selectedResourceId) {
       // this will slow things down if document gets large - need to find a better solution...
       const documents = parseAllDocuments(newValue);
-      setValid(documents.length > 0 && !documents.some(d => !isValidResourceDocument(d)));
+      // only accept single document changes for now
```
Comment (devcatalin) [inline src/components/molecules/Monaco/Monaco.tsx:191]: does editing multi-resource files still work in this case? if not, should the editor be read-only?

**Review by devcatalin — CHANGES_REQUESTED** [review]

File: src/redux/services/resource.ts (hunk 1 above)
Comment (olensmar) [reply src/redux/services/resource.ts:461]: good idea!

File: src/components/molecules/Monaco/Monaco.tsx (hunk 2 above)
Comment (olensmar) [reply src/components/molecules/Monaco/Monaco.tsx:191]: i'll double-check, but yes - selectedResourceId should not be set if the user is editing a multiresource file. This specific case is only to ensure that a user doesn't try to turn a single resources into multiple resources by manually inserting a resource delimiter (which we could support in the future)

**Review by devcatalin — APPROVED** [review]
Nice!

**Pull request merged**